apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles/notices.js

```diff
@@ -0,0 +1,186 @@
+/* global wpcomGlobalStyles */
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { ExternalLink, Notice } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { createInterpolateElement, render, useEffect, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { useCanvas } from './use-canvas';
+import { useGlobalStylesConfig } from './use-global-styles-config';
+import './notice.scss';
+
+const trackEvent = ( eventName, isSiteEditor = true ) =>
+	recordTracksEvent( eventName, {
+		context: isSiteEditor ? 'site-editor' : 'post-editor',
+		blog_id: wpcomGlobalStyles.wpcomBlogId,
+	} );
+
+function GlobalStylesWarningNotice() {
+	const { globalStylesInUse } = useGlobalStylesConfig();
+
+	useEffect( () => {
+		if ( globalStylesInUse ) {
+			trackEvent( 'calypso_global_styles_gating_notice_view_canvas_show' );
+		}
+	}, [ globalStylesInUse ] );
+
+	if ( ! globalStylesInUse ) {
+		return null;
+	}
+
+	return (
+		<Notice status="warning" isDismissible={ false } className="wpcom-global-styles-notice">
+			{ createInterpolateElement(
+				__(
+					'Your site includes customized styles that are only visible to visitors after <a>upgrading to the Premium plan or higher</a>.',
```

Comment:
I think that's worth trying! Maybe in a follow-up?

Reply:
@BogdanUngureanu I was about to implement this on a new PR, but I'm having second thoughts now. The notice would end up with two competing links (one for upgrading, and one for opening the Styles sidebar) since we don't have any good way of giving more visual priority to one of them. This is one of the reasons why we removed the "reset styles" link we had originally (see https://github.com/Automattic/dotcom-forge/issues/1221).